plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt

```diff
@@ -167,7 +168,15 @@ class Yarn2(override val descriptor: PluginDescriptor = Yarn2Factory.descriptor,
             scopes.forEach { scope ->
                 val dependencyNames = packageJson.getDependenciesForScope(scope)
                 val dependencies = packageInfo.children.dependencies
-                    .map { packageInfoForLocator.getValue(it.locator) }
+                    .map { dependency ->
+                        // Map virtual package locators to their corresponding real package locators as only their
+                        // package infos contain the transitive dependencies.
+                        val locator = virtualPackageRegex.find(dependency.locator)?.let {
+                            "${it.groupValues[1]}@npm:${it.groupValues[2]}"
+                        } ?: dependency.locator
+
+                        packageInfoForLocator.getValue(locator)
+                    }
```

Comment:
IIUC This change only works for "direct" dependencies. 
A similar logic needs to be put into `Yarn2DependencyHandler.dependenciesFor()`. 

How about taking above mentioned function, and define it as extension property `PackageInfo.realLocator` next to the other ones in `Yarn2DependencyHandler` ?

Ideally, the test should be enhanced to have such virtual package as non-direct dependency too.